bodo/pandas/utils.py

```diff
@@ -817,9 +818,36 @@ def fallback_wrapper(attr):
     ):
 
         def silenced_method(*args, **kwargs):
+            msg = ""
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore", category=BodoLibFallbackWarning)
-                return attr(*args, **kwargs)
+                try:
+                    return attr(*args, **kwargs)
+                except TypeError as e:
+                    msg = e
+                    pass
+
+            # In some cases, fallback fails and raises TypeError due to some operations being unsupported between PyArrow types.
+            # Below logic processes deeper fallback that converts problematic PyArrow types to their Pandas equivalents.
+            warnings.warn(
+                BodoLibFallbackWarning(
+                    "TypeError triggering deeper fallback. Converting PyarrowDtype elements in self to Pandas dtypes."
+                )
+            )
+            # TODO: generalize this. Currently, this method is only used for BodoSeries.
+            pd_self = pd.Series(self)
```

Comment:
Wrap this in an if checking self for BodoSeries and in the else raise exception.